### PR TITLE
testing framework - generate mandos

### PR DESCRIPTION
**Pull request by dorin-iancu**
- generate mandos output for set blockchain/accounts state
- generate mandos output for SCCalls and SCQueries
- use relative paths in generated mandos

### CODE REVIEW

File: elrond-wasm-debug/src/testing_framework/contract_obj_wrapper.rs [hunk 1]
```diff
@@ -35,16 +41,30 @@ where
     CB: ContractBase<Api = DebugApi> + CallableContract<DebugApi> + 'static,
     ContractObjBuilder: 'static + Copy + Fn(DebugApi) -> CB,
 {
-    pub fn new(_workspace_path: &str) -> Self {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        let mut current_dir = std::env::current_dir().unwrap();
+        current_dir.push(PathBuf::from_str("mandos/").unwrap());
```
Comment (andrei-marinica) [inline elrond-wasm-debug/src/testing_framework/contract_obj_wrapper.rs:47]: I think the final slash is not necessary, but doesn't hurt either

File: elrond-wasm-debug/src/testing_framework/mandos_generator.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                name: None,
+                steps: Vec::new(),
+            },
+            current_tx_id: 0,
+        }
+    }
+
+    pub fn write_mandos_output(self, file_path: &str) {
+        let buf = Vec::new();
+        let formatter = serde_json::ser::PrettyFormatter::with_indent(b"    ");
```
Comment (andrei-marinica) [inline elrond-wasm-debug/src/testing_framework/mandos_generator.rs:31]: We could extract this logic somewhere, I think it is also duplicated in some other tests.

File: elrond-wasm-debug/src/testing_framework/raw_converter.rs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+pub(crate) fn u64_as_raw(value: u64) -> ValueSubTree {
+    ValueSubTree::Str(value.to_string())
+}
+
+pub(crate) fn bytes_as_raw(bytes: &[u8]) -> ValueSubTree {
+    ValueSubTree::Str(bytes_to_hex(bytes))
+}
+
+pub(crate) fn bytes_to_hex(bytes: &[u8]) -> String {
+    "0x".to_owned() + &hex::encode(bytes)
```
Comment (andrei-marinica) [inline elrond-wasm-debug/src/testing_framework/raw_converter.rs:187]: I think something like `format!("0x{}", ...)` does this more elegantly, but this is also ok

**Review by claudiu725 — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

File: elrond-wasm-debug/src/testing_framework/raw_converter.rs (hunk 3 above)
Comment (dorin-iancu) [reply elrond-wasm-debug/src/testing_framework/raw_converter.rs:187]: Good idea, always forget about format. Will fix in the next PR.

File: elrond-wasm-debug/src/testing_framework/mandos_generator.rs (hunk 2 above)
Comment (dorin-iancu) [reply elrond-wasm-debug/src/testing_framework/mandos_generator.rs:31]: I copied this from the serde test, so yeah, it's duplicated from there.

File: elrond-wasm-debug/src/testing_framework/contract_obj_wrapper.rs (hunk 1 above)
Comment (dorin-iancu) [reply elrond-wasm-debug/src/testing_framework/contract_obj_wrapper.rs:47]: iirc, PathBuff::push() doesn't work correctly without the slash, as it considers it an absolute path and overwrites the whole path with it.

**Pull request merged**